examples/CtxtFilterExample.js

```diff
@@ -0,0 +1,233 @@
+const { DataCtxt } = require('./helpers/HOC');
+const FakeObjectDataListStore = require('./helpers/FakeObjectDataListStore');
+const { Table, Column, Cell } = require('fixed-data-table-2');
+const React = require('react');
+
+class DataListWrapper {
+  constructor(data) {
+    this._data = data;
+    this._indexMap = null;
+    this._callback = null;
+  }
+
+  // The callback is used for triggering re-rendering
+  setCallback(cb) {
+    this._callback = cb;
+  }
+
+  setIndexMap(index) {
+    this._indexMap = index;
+    this._callback();
+  }
+
+  getSize() {
+    if (this._indexMap === null) {
+      return this._data.getSize();
+    }
+
+    return this._indexMap.length;
+  }
+
+  getObjectAt(index) {
+    if (this._indexMap === null) {
+      return this._data.getObjectAt(index);
+    }
+
+    return this._data.getObjectAt(
+      this._indexMap[index]
+    );
+  }
+}
+
+const DataTable = DataCtxt(Table);
+
+class FilterTable extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { data, filters, ...other } = props;
+    this.state = {
+      rawData: data,
+      filteredData: new DataListWrapper(props.data),
+      filters: Object.assign({}, filters),
+      other
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (JSON.stringify(nextProps.filters) !== JSON.stringify(this.state.filters)){
+      this.setState({
+        filters: Object.assign({}, nextProps.filters),
+      });
+      this.filter();
+    }
+  }
+
+  filter() {
+    // Get and prep filters
+    let filters = {};
+    for (let key in this.props.filters) {
+      if (this.props.filters.hasOwnProperty(key) &&
+          this.props.filters[key] !== ''){
+        filters[key] = this.props.filters[key];
+      }
+    }
+    Object.keys(filters).map((key) => {
+      filters[key] = filters[key].toLowerCase();
+      return (key);
+    });
+
+    const match = (haystack, needle) =>
+      haystack.toLowerCase().indexOf(needle) !== -1;
+
+    if (Object.keys(filters).length > 0) {
+      const filteredIndexes = [];
+      for (let index = 0; index < this.state.rawData.getSize(); index += 1) {
+        const row = this.state.rawData.getObjectAt(index);
+
+        // Loop through all the filters and check if there's a match
+        const found = Object.keys(filters)
+          .map((varName) => {
+            const value = row[varName];
+
+            // If we have a set of values e.g. an array
+            //  If you're using immutablejs then you can just use List
+            //  This can be useful when you have more complex matches
+            if (value instanceof Array) {
+              const matches = value.map(x => match(x, filters[varName])).filter(x => x === true);
+              // If all filters were identified then set this to true
+              return (matches.length !== value.length);
+            }
+
+            return (!match(value, filters[varName]));
+          })
+          .filter(x => x === true)
+          .length === 0;
```

Comment:
Yeah, this should probably be improved in some clever way. I've focused on getting it to work, a for loop with a `break` should be smarter. I'm just used from R to avoid all for-loops and I'm not that familiar with the ups/downs of for-loops in JS.